src/keys/ed25519.js

```diff
@@ -1,24 +1,59 @@
 'use strict'
 
-require('node-forge/lib/ed25519')
-const forge = require('node-forge/lib/forge')
-exports.publicKeyLength = forge.pki.ed25519.constants.PUBLIC_KEY_BYTE_LENGTH
-exports.privateKeyLength = forge.pki.ed25519.constants.PRIVATE_KEY_BYTE_LENGTH
+const ed = require('noble-ed25519')
 
-exports.generateKey = async function () { // eslint-disable-line require-await
-  return forge.pki.ed25519.generateKeyPair()
+exports.publicKeyLength = 32
+exports.privateKeyLength = 64 // private key is 32 bytes actual private key and 32 bytes public key, for historical reasons
+
+exports.generateKey = async function () {
+  // the actual private key (32 bytes)
+  const privateKeyRaw = ed.utils.randomPrivateKey()
+  const publicKey = await ed.getPublicKey(privateKeyRaw)
+
+  // concatenated the public key to the private key
+  const privateKey = concatKeys(privateKeyRaw, publicKey)
+
+  return {
+    privateKey,
+    publicKey
+  }
 }
 
 // seed should be a 32 byte uint8array
 exports.generateKeyFromSeed = async function (seed) { // eslint-disable-line require-await
-  return forge.pki.ed25519.generateKeyPair({ seed })
+  if (seed.length !== 32) {
+    throw new TypeError('"seed" must be 32 bytes in length.')
+  } else if (!(seed instanceof Uint8Array)) {
+    throw new TypeError('"seed" must be a node.js Buffer, or Uint8Array.')
+  }
+
+  // based on node.forges algorithm, the seed is used directly as private key
+  const privateKeyRaw = seed
+  const publicKey = await ed.getPublicKey(privateKeyRaw)
+
+  const privateKey = concatKeys(privateKeyRaw, publicKey)
+
+  return {
+    privateKey,
+    publicKey
+  }
+}
+
+exports.hashAndSign = function (privateKey, msg) { // eslint-disable-line require-await
+  const privateKeyRaw = privateKey.slice(0, 32)
```

Comment:
should we make above export for key length constants and replace 32 here and in concatKeys by it?